src/test/java/core/basesyntax/service/WriterTest.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.impl.WriterCsvImpl;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class WriterTest {
+    private static Writer writer;
+    private static final String VALID_OUTPUT_DATA_PATH = "src/test/resources/validOutputData.csv";
+    private static final String VALID_REPORT =
+            "fruit,quantity" + System.lineSeparator()
+                    + "banana,152" + System.lineSeparator()
+                    + "apple,90" + System.lineSeparator();
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        writer = new WriterCsvImpl();
+    }
+
+    @Test
+    public void writeToFile_validDataAndPath_ok() {
+        writer.writeToFile(VALID_REPORT, VALID_OUTPUT_DATA_PATH);
+        String expected = readFromFile(VALID_OUTPUT_DATA_PATH);
+        Assert.assertEquals(expected, VALID_REPORT);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void writeToFile_validDataAndNullPath_notOk() {
+        writer.writeToFile(VALID_REPORT, null);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void writeToFile_nullDataAndNullPath_notOk() {
+        writer.writeToFile(null, null);
+    }
```

Comment:
Is that method notOk cause of data is null or cause of Path is null?

Reply:
Both causes - null data and null path